The job **`x86_64-gnu-llvm-15`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/6302166688/job/17108783652) [(plain)](https://github.com/rust-lang/rust/commit/204272b86d6faae84944fdaff67169c5a32bd2f3/checks/17108783652/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

   Doc-tests rustc_lint_defs

running 128 tests
ii.....i...i.........i...F....i.ii...i.......i.......iii...i..................i.........  88/128

failures:

---- compiler/rustc_lint_defs/src/builtin.rs - builtin::CONST_PATTERNS_WITHOUT_PARTIAL_EQ (line 2319) stdout ----
---- compiler/rustc_lint_defs/src/builtin.rs - builtin::CONST_PATTERNS_WITHOUT_PARTIAL_EQ (line 2319) stdout ----
warning: unknown lint: `constant_patterns_without_partial_eq`
##[warning] --> compiler/rustc_lint_defs/src/builtin.rs:2320:9
  |
2 | #![deny(constant_patterns_without_partial_eq)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: did you mean: `const_patterns_without_partial_eq`
  = note: `#[warn(unknown_lints)]` on by default


warning: to use a constant of type `EnumSet<Enum8>` in a pattern, the type must implement `PartialEq`
   |
   |
22 |         CONST_SET => { /* ok */ }
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #116122 <https://github.com/rust-lang/rust/issues/116122>
   = note: for more information, see issue #116122 <https://github.com/rust-lang/rust/issues/116122>
   = note: `#[warn(const_patterns_without_partial_eq)]` on by default
warning: 2 warnings emitted

Test compiled successfully, but it's marked `compile_fail`.


failures:
    compiler/rustc_lint_defs/src/builtin.rs - builtin::CONST_PATTERNS_WITHOUT_PARTIAL_EQ (line 2319)

test result: FAILED. 109 passed; 1 failed; 18 ignored; 0 measured; 0 filtered out; finished in 980.09ms

error: doctest failed, to rerun pass `-p rustc_lint_defs --doc`
  local time: Mon Sep 25 16:51:01 UTC 2023
  network time: Mon, 25 Sep 2023 16:51:01 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/compiler/rustc_lint_defs/src/builtin.rs
+++ b/compiler/rustc_lint_defs/src/builtin.rs
@@ -2311,13 +2311,13 @@ declare_lint! {
 }
 
 declare_lint! {
-    /// The `match_without_partial_eq` lint detects constants that are used in patterns,
+    /// The `const_patterns_without_partial_eq` lint detects constants that are used in patterns,
     /// whose type does not implement `PartialEq`.
     ///
     /// ### Example
     ///
     /// ```rust,compile_fail
-    /// #![deny(match_without_partial_eq)]
+    /// #![deny(const_patterns_without_partial_eq)]
     ///
     /// trait EnumSetType {
     ///    type Repr;
@@ -2352,12 +2352,12 @@ declare_lint! {
     /// field-by-field. In the future we'd like to ensure that pattern matching always
     /// follows `PartialEq` semantics, so that trait bound will become a requirement for
     /// matching on constants.
-    pub MATCH_WITHOUT_PARTIAL_EQ,
+    pub CONST_PATTERNS_WITHOUT_PARTIAL_EQ,
     Warn,
     "constant in pattern does not implement `PartialEq`",
     @future_incompatible = FutureIncompatibleInfo {
         reason: FutureIncompatibilityReason::FutureReleaseErrorReportInDeps,
-        reference: "issue #X <https://github.com/rust-lang/rust/issues/X>",
+        reference: "issue #116122 <https://github.com/rust-lang/rust/issues/116122>",
     };
 }
 
@@ -3407,6 +3407,7 @@ declare_lint_pass! {
         CONFLICTING_REPR_HINTS,
         CONST_EVALUATABLE_UNCHECKED,
         CONST_ITEM_MUTATION,
+        CONST_PATTERNS_WITHOUT_PARTIAL_EQ,
         DEAD_CODE,
         DEPRECATED,
         DEPRECATED_CFG_ATTR_CRATE_TYPE_NAME,
@@ -3440,7 +3441,6 @@ declare_lint_pass! {
         LOSSY_PROVENANCE_CASTS,
         MACRO_EXPANDED_MACRO_EXPORTS_ACCESSED_BY_ABSOLUTE_PATHS,
         MACRO_USE_EXTERN_CRATE,
-        MATCH_WITHOUT_PARTIAL_EQ,
         META_VARIABLE_MISUSE,
         MISSING_ABI,
         MISSING_FRAGMENT_SPECIFIER,

--- a/compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs
+++ b/compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs
@@ -246,7 +246,7 @@ impl<'tcx> ConstToPat<'tcx> {
             // any errors.) This ensures it shows up in cargo's future-compat reports as well.
             if !self.type_has_partial_eq_impl(cv.ty()) {
                 self.tcx().emit_spanned_lint(
-                    lint::builtin::MATCH_WITHOUT_PARTIAL_EQ,
+                    lint::builtin::CONST_PATTERNS_WITHOUT_PARTIAL_EQ,
                     self.id,
                     self.span,
                     NonPartialEqMatch { non_peq_ty: cv.ty() },

--- a/tests/ui/consts/const_in_pattern/issue-65466.stderr
+++ b/tests/ui/consts/const_in_pattern/issue-65466.stderr
@@ -5,8 +5,8 @@ LL |         C => (),
    |         ^
    |
    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
-   = note: for more information, see issue #X <https://github.com/rust-lang/rust/issues/X>
-   = note: `#[warn(match_without_partial_eq)]` on by default
+   = note: for more information, see issue #116122 <https://github.com/rust-lang/rust/issues/116122>
+   = note: `#[warn(const_patterns_without_partial_eq)]` on by default
 
 warning: 1 warning emitted
 
@@ -18,6 +18,6 @@ LL |         C => (),
    |         ^
    |
    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
-   = note: for more information, see issue #X <https://github.com/rust-lang/rust/issues/X>
-   = note: `#[warn(match_without_partial_eq)]` on by default
+   = note: for more information, see issue #116122 <https://github.com/rust-lang/rust/issues/116122>
+   = note: `#[warn(const_patterns_without_partial_eq)]` on by default
 

--- a/tests/ui/match/issue-72896-non-partial-eq-const.stderr
+++ b/tests/ui/match/issue-72896-non-partial-eq-const.stderr
@@ -5,8 +5,8 @@ LL |         CONST_SET => { /* ok */ }
    |         ^^^^^^^^^
    |
    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
-   = note: for more information, see issue #X <https://github.com/rust-lang/rust/issues/X>
-   = note: `#[warn(match_without_partial_eq)]` on by default
+   = note: for more information, see issue #116122 <https://github.com/rust-lang/rust/issues/116122>
+   = note: `#[warn(const_patterns_without_partial_eq)]` on by default
 
 warning: 1 warning emitted
 
@@ -18,6 +18,6 @@ LL |         CONST_SET => { /* ok */ }
    |         ^^^^^^^^^
    |
    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
-   = note: for more information, see issue #X <https://github.com/rust-lang/rust/issues/X>
-   = note: `#[warn(match_without_partial_eq)]` on by default
+   = note: for more information, see issue #116122 <https://github.com/rust-lang/rust/issues/116122>
+   = note: `#[warn(const_patterns_without_partial_eq)]` on by default
 

